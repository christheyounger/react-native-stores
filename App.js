import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import createSagaMiddleware from 'redux-saga'
import reducer from './app/reducers';
import sagas from './app/sagas';
import StoresList from './app/containers/stores';
import StoreDetails from './app/containers/storeDetails';
import { createStackNavigator, createAppContainer } from 'react-navigation';

const MainNavigator = createStackNavigator({
  Home: StoresList,
  StoreDetails: StoreDetails,
});

const Navigation = createAppContainer(MainNavigator);

const sagaMiddleware = createSagaMiddleware();


export default class App extends React.Component {
  constructor(props){
    super(props);
    this.store = createStore(reducer, applyMiddleware(sagaMiddleware));
    sagaMiddleware.run(sagas);
  }
  render() {
    return (
      <Provider store={this.store}>
        <Navigation />
      </Provider>
    );
  }
}
