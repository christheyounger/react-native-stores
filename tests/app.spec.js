
import React from 'react';
import { View } from "react-redux";
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import App from '../App';

const mockStore = configureStore();
const store = mockStore({
    user: {
      isLoaded: true,
    },
  });

jest.mock('react-navigation', () => ({
    createStackNavigator: () => ({}),
    createAppContainer: component => component,
}));

jest.mock('react-redux', () => ({
    createStore: () => mockStore({}),
    Provider: () => () => 'View'
}));

jest.mock('../app/containers/stores', () => () => 'View');
jest.mock('../app/containers/storeDetails', () => () => 'View');

describe('main app', () => {
    test('render', () => {
        const wrapper = renderer.create(<App/>);
        expect(wrapper).toMatchSnapshot();
    });
});