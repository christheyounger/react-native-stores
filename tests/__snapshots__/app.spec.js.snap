// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main app render 1`] = `null`;
